CI failure: Deploy — credential configuration conflict with --no-creds flag

## CI Failure Report

**Workflow:** Deploy
**Branch:** main
**Commit:** 38f902f798318fcd758e594dc3ae6aab384bee45
**Run:** https://github.com/Action-Llama/agents/actions/runs/23414153350

## Failure Summary

The Deploy workflow is failing because the `npx al push` command is run with the `--no-creds` flag, which prevents it from using the credentials that were just configured in the previous step.

## Failed Job(s)

### deploy

**Error output:**
```
Deploy	2026-03-22T22:35:39.7817503Z Credential error: 3 credential(s) missing: github_token, git_ssh, anthropic_key.
Deploy	2026-03-22T22:35:39.7818620Z   [MISSING] GitHub Personal Access Token (github_token)
Deploy	2026-03-22T22:35:39.7820130Z Run 'al doctor' interactively to configure them.
Deploy	2026-03-22T22:35:39.7820874Z   [MISSING] SSH Key & Git Identity (git_ssh)
Deploy	2026-03-22T22:35:39.7821887Z   [MISSING] Anthropic API Credential (anthropic_key)
Deploy	2026-03-22T22:35:39.8053461Z ##[error]Process completed with exit code 1.
```

## Root Cause Analysis

The workflow configures credentials in step 7 "Configure credentials for action-llama" by creating JSON files in `~/.action-llama/credentials/`:
- github_token.json
- git_ssh.json
- anthropic_key.json

However, the Deploy step runs `npx al push --env prod --headless --no-creds`. The `--no-creds` flag explicitly tells the action-llama tool to NOT use any credentials, causing it to report that all 3 credentials are missing.

## Suggested Fix

Remove the `--no-creds` flag from the deploy command in `.github/workflows/deploy.yml` line 51:

```diff
- run: npx al push --env prod --headless --no-creds
+ run: npx al push --env prod --headless
```

## Additional Context

The most recent commit (38f902f) was supposed to "fix: Configure action-llama credentials properly in deploy workflow" but it added the credential configuration without removing the `--no-creds` flag that prevents those credentials from being used.

---
*This issue was automatically created by the gh-actions-responder agent.*

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -58,4 +58,4 @@ jobs:
           GITHUB_TOKEN: ${{ github.token }}
           ANTHROPIC_KEY: dummy
           GIT_SSH_COMMAND: "ssh -o StrictHostKeyChecking=no"
-        run: npx al push --env prod --headless --no-creds
+        run: npx al push --env prod --headless

